Source/WebKit/Platform/IPC/unix/ConnectionUnix.cpp

```diff
@@ -96,11 +96,21 @@ class AttachmentInfo {
 
 static_assert(sizeof(MessageInfo) + sizeof(AttachmentInfo) * attachmentMaxAmount <= messageMaxSize, "messageMaxSize is too small.");
 
-void Connection::platformInitialize(Identifier identifier)
+int Connection::socketDescriptor() const
 {
-    m_socketDescriptor = identifier.handle;
 #if USE(GLIB)
-    m_socket = adoptGRef(g_socket_new_from_fd(m_socketDescriptor, nullptr));
+    return g_socket_get_fd(m_socket.get());
+#else
+    return m_socketDescriptor.value();
+#endif
+}
+
+void Connection::platformInitialize(Identifier&& identifier)
+{
+#if USE(GLIB)
+    m_socket = adoptGRef(g_socket_new_from_fd(identifier.handle.release(), nullptr));
```

Comment:
Actually I don't like this, (a) because errors are silently ignored, and (b) especially because on error, the fd is leaked. I will change this to release assert there is no error before landing.